src/configuration/types.rs

```diff
@@ -3,6 +3,20 @@ use dprint_core::generate_str_to_from;
 use serde::Deserialize;
 use serde::Serialize;
 
+#[derive(Clone, PartialEq, Copy, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum PreferHanging {
+  /// Always prefer multi-line indentation
+  Never,
+  /// Prefer hanging indentation for sequences with only a single item, but if there are multiple
+  /// items then use multi-line indentation
+  OnlySingleItem,
+  /// Always prefer hanging indentation
+  Always,
+}
+
+generate_str_to_from![PreferHanging, [Never, "never"], [OnlySingleItem, "onlySingleItem"], [Always, "always"]];
```

Comment:
I think let's just make this not backwards compatible. I removed true and false here.